### PR TITLE
asserts: sign and check tests and fixes

### PR DESCRIPTION
**Pull request by pedronis**
two things here:
* fix BuildAndSign so that the result can be Decode()d, headers are '\n' separated, not terminated
  we were getting an extra '\n' at the end of header that would confuse Decode()
* tests about the main unhappy paths in Check()

### CODE REVIEW

File: asserts/database_test.go [hunk 1]
```diff
@@ -118,3 +122,73 @@ func (dbs *databaseSuite) TestGenerateKey(c *C) {
...
+func (chks *checkSuite) TestCheckNoPubKey(c *C) {
+	cfg := &asserts.DatabaseConfig{Path: chks.rootDir}
+	db, err := asserts.OpenDatabase(cfg)
+	c.Assert(err, IsNil)
+
+	err = db.Check(chks.a)
+	c.Assert(err, ErrorMatches, "no valid known public key verifies assertion")
+}
+
+func (chks *checkSuite) TestCheckForgery(c *C) {
```
Comment (mvo5) [inline asserts/database_test.go]: Nice test!

**Comment by mvo5** [comment]
:+1:

File: asserts/database_test.go (hunk 1 above)
Comment (pedronis) [inline asserts/database_test.go]: @mvo5 thanks, was a bit of a pain to work out actually :)

**Comment by chipaca** [comment]
:+1:

**Pull request merged**